uwblueprint/paramedics-react
# Event/CCP flow via maps

[Notion ticket](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3d4d369c668e4d2581a2517666768b44&p=82c893cef178429583046605fd4e9535)

[Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=3767%3A0)

## Changes
- Adding and editing CCPs and events will now use the Map component. This ticket was mostly migrating queries and mutations into the `MapPage` component and rendering fields accordingly
- Removed `EventCreation` and `CCPForm` components as all functions would have been covered by `MapPage`
- Disabled editing and deleting `CCP` and `EVENT` pins on the map (must go through flow)
- Removed address placeholders throughout app and used pin address instead
- Added a much cleaner date picker for event flows
- Cascade deletion of location pins when deleting CCPs

## Screenshots
Adding events (with date picker open)
<img width="951" alt="Screen Shot 2020-12-31 at 4 25 43 PM" src="https://user-images.githubusercontent.com/21224282/103430779-d5d1e100-4b84-11eb-923a-be78fea774c4.png">

Editing events
<img width="947" alt="Screen Shot 2020-12-31 at 4 26 43 PM" src="https://user-images.githubusercontent.com/21224282/103430797-f9952700-4b84-11eb-9204-2618e7d4c540.png">

Adding CCP
<img width="949" alt="Screen Shot 2020-12-31 at 4 27 34 PM" src="https://user-images.githubusercontent.com/21224282/103430811-17628c00-4b85-11eb-8a2a-bc3faffc20ac.png">

Editing CCP
<img width="944" alt="Screen Shot 2020-12-31 at 4 27 59 PM" src="https://user-images.githubusercontent.com/21224282/103430817-26e1d500-4b85-11eb-95c4-b00093eedf7c.png">

Event dashboard now has the event address
<img width="511" alt="Screen Shot 2020-12-31 at 4 28 18 PM" src="https://user-images.githubusercontent.com/21224282/103430819-319c6a00-4b85-11eb-9c17-b7343c5f1398.png">

So does the CCP dashboard
<img width="767" alt="Screen Shot 2020-12-31 at 4 28 42 PM" src="https://user-images.githubusercontent.com/21224282/103430822-40831c80-4b85-11eb-8c01-4d09906c7d11.png">

## Testing
- Add an event -> check the event map and ensure that the event pin is there
- Edit the event -> make sure the changes persist in the event description in the event dashboard and map
- Delete the event
- Do everything above but for CCPs
- Go through the same flow for regular pins (i.e. add a pin on the map itself) and ensure you can edit and delete the pin on the map itself

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
